tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx

```diff
@@ -0,0 +1,85 @@
+// Author: Vincenzo Eduardo Padulano CERN/UPV, Ivan Kabadzhov CERN  06/2022
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RDF_RDATASETSPEC
+#define ROOT_RDF_RDATASETSPEC
+
+#include <string>
+#include <vector>
+#include <limits>
+#include <utility>
+
+#include <RtypesCore.h>               // Long64_t
+#include <ROOT/InternalTreeUtils.hxx> // ROOT::Internal::TreeUtils::RFriendInfo
+
+namespace ROOT {
+
+namespace Detail {
+namespace RDF {
+class RLoopManager;
+} // namespace RDF
+} // namespace Detail
+
+namespace RDF {
+
+namespace Experimental {
+
+class RDatasetSpec {
+
+   friend class ROOT::Detail::RDF::RLoopManager;
+
+public:
+   struct REntryRange {
+      Long64_t fStartEntry{0};
+      Long64_t fEndEntry{std::numeric_limits<Long64_t>::max()};
+      REntryRange();
```

Comment:
The default constructor is not needed, we already have default values for the data members

Reply:
but if we have the empty ctor, we can allow the user to pass `entryRange={}` as and argument to the spec - might be useful especially if we end up having more arguments after the entry range.

Reply:
Ok, then make it `default` and remove the unneeded definition in the source file :+1:

Reply:
hmm you mean this -- `REntryRange() = default;`, right?